qiita_pet/static/js/qiita.js

```diff
@@ -115,6 +140,36 @@ function show_hide_process_list() {
   }
 }
 
+/*
+ * send_samples_to_analysis send the selected samples for the given artifact ids to analysis
+ *
+ * @param aids: A list of artifact ids to add
+ * @param samples: The ids of the samples to add
+ *
+ * Note that we have a list of artifact ids cause the user can select one single
+ * artifact to add or all study artifacts
+ */
+function send_samples_to_analysis(button, aids) {
+  button.value = 'Adding';
+  button.disabled = true;
+  $(button).addClass("btn-info");
+  bootstrapAlert('We are adding ' + aids.length + ' artifact(s) to the analysis. This ' +
+                 'might take some time based on the number of samples on each artifact.', "warning", 10000);
+  $.get('/artifact/samples/', {ids:aids})
```

Comment:
Samples are added using a `get`? That is confusing... `get` should be use for read-only operations.

Reply:
that's not actually adding the samples, it's requesting the sample names from the artifacts to add. The addition is actually done in this line: `qiita_websocket.send('sel', data['data']);`

Reply:
Thanks for the explanation!